app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/DataBrokerEngine.kt

```diff
@@ -22,23 +22,31 @@ package org.eclipse.kuksa.testapp.databroker
 import org.eclipse.kuksa.CoroutineCallback
 import org.eclipse.kuksa.DataBrokerConnection
 import org.eclipse.kuksa.PropertyObserver
+import org.eclipse.kuksa.VssSpecificationObserver
 import org.eclipse.kuksa.model.Property
 import org.eclipse.kuksa.proto.v1.KuksaValV1
-import org.eclipse.kuksa.proto.v1.KuksaValV1.GetResponse
+import org.eclipse.kuksa.proto.v1.KuksaValV1.SetResponse
 import org.eclipse.kuksa.proto.v1.Types.Datapoint
 import org.eclipse.kuksa.testapp.model.ConnectionInfo
+import org.eclipse.kuksa.vsscore.model.VssSpecification
 
 interface DataBrokerEngine {
     var dataBrokerConnection: DataBrokerConnection?
 
     fun connect(connectionInfo: ConnectionInfo, callback: CoroutineCallback<DataBrokerConnection>)
-    fun fetchProperty(property: Property, callback: CoroutineCallback<GetResponse>)
-    fun updateProperty(
+    suspend fun fetchProperty(property: Property): KuksaValV1.GetResponse?
```

Comment:
FYI for now: The java user can still call it like this (it will still be auto completed): 

```
dataBrokerConnection.fetchSpecification(specification, new CoroutineCallback<VssSpecification>() {
            @Override
            public void onSuccess(@androidx.annotation.Nullable VssSpecification result) {

            }

            @Override
            public void onError(@NonNull Throwable error) {

            }
        });
```

<img width="739" alt="Screenshot 2023-09-25 at 15 37 42" src="https://github.com/eclipse-kuksa/kuksa-android-sdk/assets/1355532/a1fac856-7bb2-48f7-ba7c-596ef61a4dec">

The main difference is that the required type is now a higher interface instead of the CoroutineCallback. But since we are providing the CoroutineCallback class it may be fine?
<img width="770" alt="Screenshot 2023-09-25 at 15 38 59" src="https://github.com/eclipse-kuksa/kuksa-android-sdk/assets/1355532/9246bad8-d34b-45fd-981b-360bf92f71df">

Reply:
If we remove the suspend from the DataBrokerEngine interface and call it as you showed it would be fine for me on first glance. However I could imagine that the "VssSpecification" type used in the CoroutineCallback could create confusion?

Could you show a full example for the snippet?

Reply:
Reverted it back to the old variant so we can ensure the java compatibility. For now we gotta live with the not 100% sugar Kotlin variant until we maybe provide a JavaActivity. Fine for now.